mantidproject/mantid
# Bug fix for contour plotting issue when using a WorkspaceGroup

Fixes #19419

**Context**
When using the `Plot Contour from Group` functionality, the contour was displayed with an offset. The reason was a mismatched index for histogram data in `MantidMatrixFunction`. `MantidMatrixFunction::indexX(size_t row, double xValue)` was extended to handle point data and Matrix data correctly.

Note that there is still an offset which is "correct" and is a result of the sampling which is being used when displaying `MantidMatrix`-based workspaces. The sampling was increase by a factor of 7 which produces visually nice contour lines on a 21 inch screen.

**To test:**

You can create a Group Workspace with the script below
```python
base = CreateSampleWorkspace(Function='One Peak', NumBanks=1, BankPixelWidth=1)

wsNames = []
for t in range (10,40,10):
    wsName = "temp" + str(t)
    wsNames.append(wsName)
    CloneWorkspace(InputWorkspace=base, OutputWorkspace=wsName)
    #MultiplyRange(base,75,99,t/100.0,OutputWorkspace=wsName)
    AddSampleLog(wsName,"Temp",str(t),"Number","Kelvin")
TestGroup = GroupWorkspaces(wsNames)
TestGroup = Rebin(TestGroup, "0, 4000, 20000")
```
Confirm that when using `Plot Contour from Group` we don't get something which is significantly offset.


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
